### PR TITLE
providers/vault: add generation of CAs

### PR DESCRIPTION
**Pull request by edganiukov**
Ref NSL-3301

### CODE REVIEW

File: internal/providers/vault/ca.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	"namespacelabs.dev/foundation/internal/fnerrors"
+	"namespacelabs.dev/foundation/std/cfg"
+	"namespacelabs.dev/foundation/std/tasks"
+	"namespacelabs.dev/foundation/universe/vault"
+)
+
+func certificateAuthorityProvider(ctx context.Context, conf cfg.Configuration, secretId secrets.SecretIdentifier, cfg *vault.CertificateAuthority) ([]byte, error) {
+	vaultConfig, ok := GetVaultConfig(conf)
+	if !ok || vaultConfig == nil {
+		return nil, fnerrors.BadInputError("invalid vault app role configuration: missing provider configuration")
```
Comment (attilaolah) [inline internal/providers/vault/ca.go:29]: certificate authority configuration? Or you also need the app role here?

File: internal/providers/vault/ca.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+
+	if cfg.CommonName == "" {
+		return nil, fnerrors.BadInputError("invalid certificate authority configuration: missing common name")
+	}
+	if cfg.Ttl == "" {
+		return nil, fnerrors.BadInputError("invalid certificate authority configuration: missing ttl")
+	}
+
+	if _, err := time.ParseDuration(cfg.Ttl); err != nil {
+		return nil, fnerrors.BadInputError("ttl is not a valid duration")
```
Comment (attilaolah) [inline internal/providers/vault/ca.go:45]: Can you wrap the error, maybe it provides some useful details?

**Review by attilaolah — APPROVED** [review]

File: internal/providers/vault/ca.go (hunk 1 above)
Comment (edganiukov) [reply internal/providers/vault/ca.go]: copy-paste error, fixed.

File: internal/providers/vault/ca.go (hunk 2 above)
Comment (edganiukov) [reply internal/providers/vault/ca.go]: done

**Pull request merged**